src/common/xio_connection.c

```diff
@@ -3602,6 +3602,9 @@ void xio_connection_keepalive_intvl(void *_connection)
 			  connection, connection->ka.probes);
 		connection->ka.probes = 0;
 
+		/* notify the transport via the nexus */
+		xio_nexus_keepalive_timeout(connection->nexus);
```

Comment:
Why do you need it? There is a call to xio_disconnect right after which suppose to disconnect the connection and also close the transport layer. am I missing something?